src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs

```diff
@@ -207,6 +208,14 @@ private void ReplicateToDestination()
                         {
                             while (true)
                             {
+#if DEBUG
+                                if (_parent.WaitFormTest != null)
```

Comment:
What is this? I assume you need this to be certain that events happen in the proper order.
That is fine, but make sure that the name is accurate.